add example for polars dataframe read

wseaton opened the pull request:
example for feature discussed in #20

wseaton committed:
Add example for polars use with snowflake-api

wseaton commented:
@andrusha the build failures for the tracing example should be fixed (linux one looks transient). Thanks for tag teaming on this, the API is looking great!

andrusha commented:
@wseaton could you rebase over the main? Or maybe even just commit the example file with the fresh commit. Sorry for messing up your branch a bit.

andrusha committed:
Cleanup polars example deps

andrusha committed:
CI/CD: Use rustup instead of toolchain action

andrusha committed:
Relax polars example version to not hit incompatibility

Since the lib will compile first during the check, and example last, causing incompatibility
